src/app/pages/receive-tokens/receive-modal.tsx

```diff
@@ -22,6 +22,9 @@ import { useCurrentAccountStxAddressState } from '@app/store/accounts/blockchain
 export function ReceiveModal() {
   const analytics = useAnalytics();
   const navigate = useNavigate();
+  const {
+    state: { backgroundLocation },
+  } = useLocation();
```

Comment:
Would `useLocationState` be useful here?

Reply:
I saw it but just wanted to get a draft working. I will check it out and see if it improves things